### PR TITLE
Add S_Cylinder

### PR DESCRIPTION
**Pull request by gergondet**
This PR adds S_Cylinder

It also adds accessors for S_Capsule data

### CODE REVIEW

File: src/S_Object/S_Cylinder.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  double dot = normal_ * v;
+  if (dot > 0)
+  {
+    d = p2_;
+  }
+  Vector3 proj = v - normal_ * dot;
+  double proj_norm = proj.norm();
+  if(proj_norm != 0)
+  {
+    return d + proj * radius_ / proj_norm;
```
Comment (mehdi-benallegue) [inline src/S_Object/S_Cylinder.cpp:28]: shouldn't we add parentheses ?
`return d + proj * (radius_ / proj_norm);`

File: src/S_Object/S_Cylinder.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  }
+  return d;
```
Comment (mehdi-benallegue) [inline src/S_Object/S_Cylinder.cpp:30]: I usually prefer to use `else` but you decide

**Comment by mehdi-benallegue** [comment]
That seems okay for me. I will PR on this new shape being added to the test.

File: src/S_Object/S_Cylinder.cpp (hunk 2 above)
Comment (gergondet) [reply src/S_Object/S_Cylinder.cpp]: Changed in eec9adc

File: src/S_Object/S_Cylinder.cpp (hunk 1 above)
Comment (gergondet) [reply src/S_Object/S_Cylinder.cpp]: Changed in eec9adc

**Comment by gergondet** [comment]
Thanks. I will merge when all actions are ok then :)

**Pull request merged**